sdk/python/dagger/src/dagger/cli.py

```diff
@@ -0,0 +1,33 @@
+import logging
+import sys
+
+import click
+
+from .log import configure_logging
+from .server import Server
+
+
+@click.command(short_help="Entrypoint for a dagger extension")
+@click.option("-schema", is_flag=True, help="Save schema to file")
+def run(schema: bool):
+    sys.path.insert(0, ".")
+
+    try:
+        from main import server
```

Comment:
huge nit: would it be possible to instead always import the `Query` class and then do the two lines of boilerplate for `schema = ...` and `server = ...` on behalf of the user?

Reply:
I'd rather not.

1. If you look at the [integrations](https://strawberry.rocks/docs/integrations/aiohttp), you always wrap the `schema` instance in your own system so I specifically wanted to keep this consistency. We're just building a Dagger integration here. Look at the "Playground" section in the description, as I explain it there.
2. You need control of `schema` for more stuff. If you have types not referenced directly but still need them in the schema for example, you need to reference them in this `Schema` constructor. I.e., there's more options there that you may need.
3. As for `server`, the point is to be able to configure it in some way.
    - Right now you can set `Server(..., debug=True)` for example. More options may be available later.
    - You can also add hooks to the converter to be able to serialize some complex object that you may have added (`server.converter.register_unstructure_hook(...)`). You can even instantiate `dagger.Server` with your own `Converter` object if needed.[^1]

[^1]: If this converter stuff looks undocumented, it's not. We just need to direct to the actual [library](https://cattrs.readthedocs.io/en/latest/usage.html) that does this.

As for this import, in case you think this is hacky, in Python it's actually not. It's pretty standard. It's normal in Python to have some config (i.e., "main:server" from file, env or cli arg) to get something like this. There's primitives to get an object from a string but for now I decided the more simpler approach (static), but we may allow it to be configurable if we want to (i.e., tell us where your `dagger.Server` instance is located).

Reply:
Oh the only reason I asked here was to see if you thought it was easy to trim a little bit of boilerplate from the code the user has to write. If that constrains us in a bunch of other ways it's not a big deal at all, makes sense that we should continue letting them customize those fields (and stick close to the strawberry conventions when possible).